GitTfs/Core/GitRepository.cs

```diff
@@ -520,18 +528,23 @@ public bool CreateBranch(string gitBranchName, string target)
             return reference != null;
         }
 
-        private readonly Dictionary<long, string> changesetsCache = new Dictionary<long, string>();
+        private readonly Dictionary<long, IList<string>> changesetsCache = new Dictionary<long, IList<string>>();
         private bool cacheIsFull = false;
 
         public string FindCommitHashByChangesetId(long changesetId, string tfsPath)
         {
-            var commit = FindCommitByChangesetId(changesetId, tfsPath);
+            var commit = FindCommitsByChangesetId(changesetId).FirstOrDefault(c => c.Message.Contains(tfsPath));
```

Comment:
>Because that's not very good in terms of performance...
Most of the cases there will be only one commit in the list, so no performance issues.